### DISCUSSION TITLE
如何对接 棱束链 对象存储 实现超低成本存储图片

### DISCUSSION THREAD

**Discussion by CodeKirin-dragon** [lsky-org/lsky-pro#452] [教程]
### 棱束链介绍

> `棱束链` 是一款云存储产品，棱束链的`对象存储`相比同款产品，数据存储仅有同行约`74%`价格，流量费用更是只有约`20%`价格，`海外传输`不收取流量费用，并且没有`数据取回`费用，再也不担心使用`对象存储`流量费用过高的问题。 官网：[`棱束链`](https://www.lingshulian.com/)

### 修改存储策略为 棱束链对象存储

> 1. 打开 [`棱束链官网`](https://www.lingshulian.com) -> [`前往个人中心`](https://console.lingshulian.com/user-center) -> 在 `功能设置` 打开API -> 复制好 `secretId` 以及 `secretKey` 请妥善保管。
>
>    ![image-20220618212157982](https://s3-us-east-1-accelerate.ossfiles.com/static/%E6%96%87%E7%AB%A0/%E5%A6%82%E4%BD%95%E4%BD%BF%E7%94%A8%E6%A3%B1%E6%9D%9F%E9%93%BE%E5%BF%AB%E9%80%9F%E6%90%AD%E5%BB%BACloudreve%E8%87%AA%E5%BB%BA%E4%BD%8E%E6%88%90%E6%9C%AC%E7%BD%91%E7%9B%98%E5%AE%9E%E8%B7%B5/%E5%9B%BE%E7%89%87/5.png)
>
> 2. 进入 `对象存储` -> [`存储桶列表`](https://console.lingshulian.com/bucket-list) -> 选择一个存储桶，或者新建一个新的存储桶，我这里选择创建一个新的桶，桶权限选择 `公共读` 。
>
> ![image-20220618212642976](https://s3-us-east-1-accelerate.ossfiles.com/static/%E6%96%87%E7%AB%A0/%E5%A6%82%E4%BD%95%E4%BD%BF%E7%94%A8%E6%A3%B1%E6%9D%9F%E9%93%BE%E5%BF%AB%E9%80%9F%E6%90%AD%E5%BB%BACloudreve%E8%87%AA%E5%BB%BA%E4%BD%8E%E6%88%90%E6%9C%AC%E7%BD%91%E7%9B%98%E5%AE%9E%E8%B7%B5/%E5%9B%BE%E7%89%87/6.png)
>
> 3. 我们点击 `桶配置` 复制好`桶信息`备用，打开`跨域设置`并且设置好`跨域规则`，如您不会设置可以按照下图进行设置：（当然不配置`跨域规则`也是可以的，如果你需要把图片引入其它网站就需要配置。）
>
> ![image-20220618213019852](https://s3-us-east-1-accelerate.ossfiles.com/static/%E6%96%87%E7%AB%A0/%E5%A6%82%E4%BD%95%E4%BD%BF%E7%94%A8%E6%A3%B1%E6%9D%9F%E9%93%BE%E5%BF%AB%E9%80%9F%E6%90%AD%E5%BB%BACloudreve%E8%87%AA%E5%BB%BA%E4%BD%8E%E6%88%90%E6%9C%AC%E7%BD%91%E7%9B%98%E5%AE%9E%E8%B7%B5/%E5%9B%BE%E7%89%87/7.png)
>
> 4. 我们回到 `Lsky Pro` 的管理面板点击右下角的`存储策略`点击`创建存储策略`
>    1. `名称`  输入`棱束链对象存储`
>
>    2. `介绍`  为空即可
>
>    3. `储存策略`  选择`AWS S3`
>
>    4. `访问域名`  输入`桶信息`里面的 `桶域名` 注意需要 `https://` 开头在输入`桶域名`
>
>    5. `AccessKeyId`  输入在`个人中心`复制的`secretId`
>
>    6. `SecretAccessKey`  输入在`个人中心`复制的 `secretKey`
>    7. `连接地址`输入`桶信息`里面的 `端点` 注意需要 `https://` 开头在输入`端点`
>    8. `区域(region)`输入`桶信息`里面的 `地域` 
>    9. `储存桶名称`输入刚刚创建或者选择的桶`名称`
>
>
> ![image-20220618214028199](https://s3-us-east-1-accelerate.ossfiles.com/static/%E6%96%87%E7%AB%A0/%E5%A6%82%E4%BD%95%E5%88%A9%E7%94%A8%E6%A3%B1%E6%9D%9F%E9%93%BE%E4%BB%A5%E5%8F%8ALsky%20Pro%EF%BC%88%E5%85%B0%E7%A9%BA%E5%9B%BE%E5%BA%8A%EF%BC%89%E6%90%AD%E5%BB%BA%E4%BD%8E%E6%88%90%E6%9C%AC%E5%9B%BE%E5%BA%8A%E5%AE%9E%E8%B7%B5/3.png)
>
>   10. 点击右下方的`确认创建` 
>

### 修改默认上传策略

> 我们打开左侧导航的 `设置`选择`默认上传策略`选择`棱束链对象存储`，在点击右下角的`保存设置`；接下来 上传的图片都会保存到`棱束链对象存储`当中。
>
>    ![image-20220618214501658](https://s3-us-east-1-accelerate.ossfiles.com/static/%E6%96%87%E7%AB%A0/%E5%A6%82%E4%BD%95%E5%88%A9%E7%94%A8%E6%A3%B1%E6%9D%9F%E9%93%BE%E4%BB%A5%E5%8F%8ALsky%20Pro%EF%BC%88%E5%85%B0%E7%A9%BA%E5%9B%BE%E5%BA%8A%EF%BC%89%E6%90%AD%E5%BB%BA%E4%BD%8E%E6%88%90%E6%9C%AC%E5%9B%BE%E5%BA%8A%E5%AE%9E%E8%B7%B5/4.png)

**Comment by hanasaki-misaki** [comment]
上传失败，不知道什么情况
